flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/DispatcherJobTest.java

```diff
@@ -59,6 +62,22 @@ public void testStatusWhenInitializing() throws Exception {
         assertJobStatus(dispatcherJob, JobStatus.INITIALIZING);
     }
 
+    @Test
+    public void testArchivedCheckpointingSettingsNotNullWhenInitializingIfCheckpointingIsEnabled()
+            throws Exception {
+        final JobGraph jobGraph = JobGraphTestUtils.singleNoOpJobGraph();
+        jobGraph.setSnapshotSettings(
+                new JobCheckpointingSettings(
+                        CheckpointCoordinatorConfiguration.builder().build(), null));
+
+        TestContext testContext = createTestContext(jobGraph);
+        DispatcherJob dispatcherJob = testContext.getDispatcherJob();
```

Comment:
nit: Out of curiosity - did you miss making these `final` or is there another reason for that?

Reply:
I just copied the core test structure from another test.